### PR TITLE
Don't start server if versions.js is missing but versioning is enabled

### LINKED ISSUE #228: Versions HTML page not created
I ran the versions script to create a snapshot version.

When I run `yarn run start`, I see my website with a new link at the top for versions. Clicking that link leads to a page not found.

screen shot 2017-11-15 at 12 49 49 pm

Upon checking the `website/pages/en` folder, I don't see a `versions.html` file.

### PR DESCRIPTION
**Pull request by endiliey**
## Motivation

This prevents user to start the server when `versions.js` is missing but versioning is enabled
#betterengineering

Closes #228 based on https://github.com/facebook/Docusaurus/issues/228#issuecomment-368748690
Related to #297 #369 

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/Docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

- [X] Start server as normal. `versions.js` exists when versioning is enabled
 ```
cd website
yarn start
```

no error - versions

- [X] Delete `versions.js` when versioning is enabled (`versions.json` exist) 

```
cd website
yarn start
```

error

### CODE REVIEW

**Comment by endiliey** [comment]
@yangshun Great idea. PR updated !
I added error property on env.versioning class.
This will simplify adding extra errors in the future as well.

File: lib/server/env.js [hunk 1]
```diff
@@ -58,6 +61,12 @@ class Versioning {
         ? siteConfig.defaultVersionShown
         : this.versions[0]; // otherwise show the latest version (other than next/master)
     }
+
+    if (!fs.existsSync(versions_js)) {
+      this.error = `${chalk.yellow(
```
Comment (yangshun) [inline lib/server/env.js]: Please split the error flag and messages into separate fields:

```js
constructor {
  ...
  this.missingVersionsPage = false;
}

printMissingVersionsPageError() {
  console.error(`${chalk.yellow(...)`);
}
```

File: lib/start-server.js [hunk 2]
```diff
@@ -34,6 +35,11 @@ if (!fs.existsSync(CWD + '/siteConfig.js')) {
   process.exit(1);
 }
 
+if (env.versioning.enabled && env.versioning.error) {
```
Comment (yangshun) [inline lib/start-server.js]:
```js
if (env.versioning.enabled && env.versioning.missingVersionsPage) {
  env.printMissingVersionsPageError();
}
```

File: lib/server/env.js (hunk 1 above)
Comment (endiliey) [reply lib/server/env.js]: done

**Comment by endiliey** [comment]
@yangshun done ! feels great to have a better & cleaner code 😂

**Pull request merged**